src/realm/set.hpp

```diff
@@ -767,6 +833,162 @@ void Set<T>::do_erase(size_t ndx)
+bool Set<T>::intersects(const Rhs& rhs) const
+{
+    return intersects(std::begin(rhs), std::end(rhs));
+}
+
+namespace _impl {
+template <class T>
+struct CountingOutputIterator {
+    using iterator_category = std::output_iterator_tag;
+    using value_type = void;
+    using difference_type = void;
+    using pointer = void;
+    using reference = void;
+
+    explicit CountingOutputIterator(size_t& num)
+        : num(num)
+    {
+    }
+
+    size_t& num;
+
+    CountingOutputIterator& operator=(const T&)
+    {
+        ++num;
+        return *this;
+    }
+
+    CountingOutputIterator& operator*()
+    {
+        return *this;
+    }
+
+    CountingOutputIterator& operator++()
+    {
+        return *this;
+    }
+};
+} // namespace _impl
+
+template <class T>
+template <class It1, class It2>
+bool Set<T>::intersects(It1 first, It2 last) const
+{
+    size_t count = 0;
+    std::set_intersection(begin(), end(), first, last, _impl::CountingOutputIterator<T>{count},
+                          SetElementLessThan<T>{});
+    return count != 0;
+}
+
+template <class T>
+template <class Rhs>
+inline void Set<T>::assign_union(const Rhs& rhs)
+{
+    assign_union(std::begin(rhs), std::end(rhs));
+}
+
+template <class T>
+template <class It1, class It2>
+void Set<T>::assign_union(It1 first, It2 last)
+{
+    std::vector<T> the_union;
+    std::set_union(begin(), end(), first, last, std::back_inserter(the_union), SetElementLessThan<T>{});
+    clear();
+    for (auto value : the_union) {
+        insert(value);
+    }
+}
+
+template <class T>
+template <class Rhs>
+inline void Set<T>::assign_intersection(const Rhs& rhs)
+{
+    assign_intersection(std::begin(rhs), std::end(rhs));
+}
+
+template <class T>
+template <class It1, class It2>
+void Set<T>::assign_intersection(It1 first, It2 last)
+{
+    std::vector<T> intersection;
+    std::set_intersection(begin(), end(), first, last, std::back_inserter(intersection), SetElementLessThan<T>{});
+    clear();
+    for (auto value : intersection) {
+        insert(value);
+    }
+}
+
+template <class T>
+template <class Rhs>
+inline void Set<T>::assign_difference(const Rhs& rhs)
+{
+    assign_difference(std::begin(rhs), std::end(rhs));
+}
+
+template <class T>
+template <class It1, class It2>
+void Set<T>::assign_difference(It1 first, It2 last)
+{
+    std::vector<T> difference;
+    std::set_difference(begin(), end(), first, last, std::back_inserter(difference), SetElementLessThan<T>{});
+    clear();
+    for (auto value : difference) {
+        insert(value);
+    }
+}
+
+template <class T>
+template <class Rhs>
+inline void Set<T>::assign_symmetric_difference(const Rhs& rhs)
+{
+    assign_symmetric_difference(std::begin(rhs), std::end(rhs));
+}
+
+template <class T>
+template <class It1, class It2>
+void Set<T>::assign_symmetric_difference(It1 first, It2 last)
+{
+    std::vector<T> difference;
+    std::set_symmetric_difference(begin(), end(), first, last, std::back_inserter(difference),
```

Comment:
You can just go through all elements in the second set. If element is present in current set then remove else add.